ivy/functional/ivy/nest.py

```diff
@@ -163,12 +175,26 @@ def set_nest_at_index(
         b: ivy.array([3., 4.])
     }
     """
+    nest_type = type(nest)
+    if _result is None:
+        if shallow:
+            _result = nest_type(nest)
+        else:
+            _result = copy_nest(nest)
+    _result = list(_result) if isinstance(nest, tuple) else _result
     if len(index) == 1:
-        nest[index[0]] = value
+        if shallow:
+            try:
+                nest[index[0]] = value
+            except TypeError:
+                print("shallow cannot be set to True for immutable types")
```

Comment:
maybe this ends up being a little too verbose but we can change it afterwards anyway